src/precompiles/modexp.rs

```diff
@@ -60,96 +88,66 @@ impl Precompile for ModExp<Byzantium> {
         let exp_len = U256::from(&input[32..64]);
         let mod_len = U256::from(&input[64..96]);
 
-        let mul = Self::mult_complexity(core::cmp::max(mod_len, base_len))?;
-        let adj = core::cmp::max(Self::adj_exp_len(exp_len, base_len, input), U256::from(1))
-            / U256::from(20);
-        let (gas_val, overflow) = mul.overflowing_mul(adj);
-        if overflow {
-            Err(ExitError::OutOfGas)
-        } else {
-            Ok(gas_val.as_u64())
-        }
+        let mul = Self::mul_complexity(core::cmp::max(mod_len, base_len))?;
+        let iter_count = Self::calc_iter_count(exp_len, base_len, &input);
+        let maybe_gas = mul.checked_mul(core::cmp::max(iter_count, U256::from(1)));
```

Comment:
Overflow check not necessary since `mul` is bounded by `2^61` and `iter_count` is bounded by `2^40`, so the whole result is bounded by `2^101 < 2^256`.